packages/compass-indexes/src/components/create-index-modal/create-index-modal.jsx

```diff
@@ -513,7 +500,7 @@ const mapStateToProps = (state) => ({
   isPartialFilterExpression: state.isPartialFilterExpression,
   partialFilterExpression: state.partialFilterExpression,
   isCustomCollation: state.isCustomCollation,
-  collation: state.collation,
+  collationString: state.collationString,
```

Comment:
Inside the state that is not really a string anyway or?

I'm not sure this renaming is necessary, to me, it doesn't make it too much clearer than before, and if the value of that is not really a string it makes it a bit confusing.

I like that the action is called `collationStringChanged` though, that is perfect, but I'd never expect `state.collationString` not to be a string.

What about we keep the module name as `collation.js`, the state as `.collation` and the action as `COLLATION_STRING_CHANGED`?

Reply:
Just for some extra context, https://github.com/mongodb-js/compass/pull/3285 would make it so that the BSON document editor component would update its user by passing the current value, not (only) the current text content, so it might be good to keep that in mind here

Reply:
I have aligned it with what we did in aggregations: https://github.com/mongodb-js/compass/blob/main/packages/compass-aggregations/src/modules/collation-string.ts

There are a couple of places already where this collation is handled differently. I think we should use one way everywhere and then improve naming.

Reply:
ah nice, does that keep the string as internal value?

Reply:
It will be a server error shown to a user since you specified a custom collation, but the validated string returned null. Something like `The field 'collation' must be an object, but got null`.

Reply:
Renamed `collationString` to `collationInfo` to reduce confusion about what it stores.

Reply:
The error in this case does look a bit weird though 🤔 I have a string value but am shown that there is an issue with collation because it was `null` which totally would prompt me to open an issue and ask why my collation is not being sent to the server

![image](https://user-images.githubusercontent.com/5036933/181704000-ebc1d144-9b3a-4200-aabd-959ae484bd66.png)

We definitely can decide that we stop doing our own validation and rely on the server, but then we should also stop modifying payloads that we pass to server commands probably, otherwise the errors can be confusing